tests_e2e/tests/agent_status/agent_status.py

```diff
@@ -0,0 +1,143 @@
+from tests_e2e.tests.lib.virtual_machine_client import VirtualMachineClient
+
+
+def validate_instance_view_vmagent_status(instance_view: VirtualMachineInstanceView):
+    is_valid = True
+    status = instance_view.vm_agent.statuses[0]
+
+    # Validate message field
+    message = status.message
+    if message is None:
+        is_valid = False
+        log.info("Instance view is missing an agent status message, waiting to retry...")
+    elif 'unresponsive' in message:
+        is_valid = False
+        log.info("Instance view shows unresponsive agent, waiting to retry...")
+
+    # Validate display status field
+    display_status = status.display_status
+    if display_status is None:
+        is_valid = False
+        log.info("Instance view is missing an agent display status, waiting to retry...")
+    elif 'Not Ready' in display_status:
+        is_valid = False
+        log.info("Instance view shows agent status is not ready, waiting to retry...")
+
+    return is_valid
+
+
+def validate_instance_view_vmagent(instance_view: VirtualMachineInstanceView):
+    is_valid = True
+
+    # Validate vm_agent_version field
+    vm_agent_version = instance_view.vm_agent.vm_agent_version
+    if vm_agent_version is None:
+        is_valid = False
+        log.info("Instance view is missing agent version, waiting to retry...")
+    elif 'Unknown' in vm_agent_version:
+        is_valid = False
+        log.info("Instance view shows agent version is unknown, waiting to retry...")
+
+    # Validate statuses field
+    statuses = instance_view.vm_agent.statuses
+    if statuses is None:
+        is_valid = False
+        log.info("Instance view is missing agent statuses, waiting to retry...")
+    elif len(statuses) < 1:
+        is_valid = False
+        log.info("Instance view is missing an agent status entry, waiting to retry...")
+    else:
+        is_valid = validate_instance_view_vmagent_status(instance_view=instance_view)
+
+    return is_valid
+
+
+def validate_instance_view(instance_view: VirtualMachineInstanceView):
+    instance_view_is_valid = True
+
+    if instance_view.vm_agent is None:
+        instance_view_is_valid = False
+        log.info("Instance view is missing vm agent, waiting to retry...")
+    else:
+        instance_view_is_valid = validate_instance_view_vmagent(instance_view=instance_view)
+
+    if instance_view.statuses is None:
+        instance_view_is_valid = False
+        log.info("Instance view is missing statuses, waiting to retry...")
+
+    if instance_view_is_valid:
+        log.info("Instance view is valid, agent version: {0}, status: {1}"
+                 .format(instance_view.vm_agent.vm_agent_version,
+                         instance_view.vm_agent.statuses[0].display_status))
+
+    return instance_view_is_valid
+
+
+class AgentStatus(AgentTest):
+    def __init__(self, context: AgentTestContext):
+        super().__init__(context)
+        self._ssh_client = SshClient(
```

Comment:
We do have method in AgentTestContext, reusing that self._context.create_ssh_client() simplifies things.